src/app.py

```diff
@@ -78,90 +81,312 @@ def index():
 """
 Show the current VERSION and BUILD, as well as the status of MySQL connection.
 
+Parameters
+----------
+None
+
 Returns
 -------
-json
-    A json containing the status details
+HTTP 200 Response with Content-Type application/json
+The JSON body is a dictionary with the status details.
 """
 # Status of MySQL connection
 @app.route('/status', methods=['GET'])
 def status():
-    global worker
+    global ukv_worker
 
-    response_data = {
-        # Use strip() to remove leading and trailing spaces, newlines, and tabs
+    # Use strip() to remove leading and trailing spaces, newlines, and tabs
+    status_data = {
         'version': (Path(__file__).absolute().parent.parent / 'VERSION').read_text().strip(),
         'build': (Path(__file__).absolute().parent.parent / 'BUILD').read_text().strip(),
-        'mysql_connection': False
+        'mysql_connection': ukv_worker.test_connection()
     }
-
-    dbcheck = worker.testConnection()
-    if dbcheck:
-        response_data['mysql_connection'] = True
-
-    return jsonify(response_data)
+    return jsonify(status_data)
 
 """
 An endpoint to create or update a key/value pair for the authenticated user.
 The value is valid JSON attached to the HTTP Request.
-On success returns an HTTP 200 Response.
 
 Parameters
 ----------
-key : The key to store the value under for the authenticated user, as specified on the API endpoint 
+key : The key to store the value under for the authenticated user, as specified on the API endpoint. The
+      key cannot contain any whitespace, the characters %, _, *, #,',",`, or the character sequences /*, */, or --.
 
 Returns
 -------
-HTTP 200 Response
+HTTP 200 Response with Content-Type application/json
+The JSON body is a dictionary with an entry confirming the key/value pair was stored.
+
+HTTP 400 Response with Content-Type application/json
+The JSON body is a dictionary with an error message describing the problem that kept
+the key/value pair from being stored.  Examples causes are
+- A body in the Request which is not valid JSON to be stored as a key's value
+- Failure to specify a Content-Type header of application/json for the Request
+- A body in the Request which is valid JSON but is empty
+- Keys which are too long
+- Keys which contain whitespace or other disallowed characters
+
+HTTP 500 Response
+An unexpected error in the server, including unexpected problems with the data store.
 """
 @app.route(rule='/user/keys/<key>', methods=["PUT"])
 @secured(has_write=True)
-def upsert_key_value(key):
-    global worker
+def upsert_key_value(key: Annotated[str, 50]):
```

Comment:
To avoid confusion, does `Annotated[str, 50]` suggest the max length is 50 or the key length needs to be exact 50?

Reply:
It is just type hinting introduced in Python 3.10 for readability, documentation, and static code analysis. Not "enforced", so the explicit verification of length remains in the code. When we were at Python 3.9, I would have just used str for the hint.